Src/Plugins/Security/SEC101_038.PostgreSqlCredentialsValidator.cs

```diff
@@ -61,6 +67,16 @@ protected override IEnumerable<ValidationResult> IsValidStaticHelper(Dictionary<
             groups.TryGetNonEmptyValue("resource", out FlexMatch resource);
 
             string hostValue = FilteringHelpers.StandardizeLocalhostName(host.Value);
+            string idValue = id.Value;
+
+            // Username must be in the form <username>@<hostname> to communicate with Azure.
+            // If the username does not contain a host name, we can't connect.
+            if (AzureHosts.Any(azHosts => hostValue.IndexOf(azHosts, StringComparison.OrdinalIgnoreCase) != -1) &&
```

Comment:
>Any [](http://example.com/codeflow?start=27&length=3)

You're side-stepping the value of having a set here, which is the ability to call Contains on it for look-up in constant time. Also, you can initialize your set with a string comparer that makes the Contains look-up case-insensitive. You may be required to refactor your initialization strategy though, since this will entail calling a parameterized ctor.

Btw - since you are only tracking a single host, an alternate approach is simply to persist this data as a const string (and to upgrade to a hashset later, if/when additional hosts require it).

Reply:
Makes sense. I was trying to keep the file self-consistent. Will swap it to `.contains`